source/extensions/omni.isaac.orbit/omni/isaac/orbit/devices/gamepad/se3_gamepad.py

```diff
@@ -0,0 +1,203 @@
+
+        A list of available keys are present in the
+        `carb documentation <https://docs.omniverse.nvidia.com/kit/docs/carbonite/latest/docs/python/carb.html?highlight=gamepadeventtype#carb.input.GamepadInput>`_.
+
+        The callback function should not take any arguments.
+
+        Args:
+            key (str): The gamepad button to check against.
+            func (Callable): The function to call when key is pressed.
+        """
+        self._additional_callbacks[key] = func
+
+    def advance(self) -> Tuple[np.ndarray, bool]:
+        """Provides the result from gamepad event state.
+
+        Returns:
+            Tuple[np.ndarray, bool] -- A tuple containing the delta pose command and gripper commands.
+        """
+        delta_rot_sgn = self._delta_rot[0, :] > self._delta_rot[1, :]
+        delta_rot = self._delta_rot.max(axis=0)
+        delta_rot[~delta_rot_sgn] *= -1
+        delta_pos_sgn = self._delta_pos[0, :] > self._delta_pos[1, :]
+        delta_pos = self._delta_pos.max(axis=0)
+        delta_pos[~delta_pos_sgn] *= -1
+
+        rot_vec = Rotation.from_euler("XYZ", delta_rot).as_rotvec()
+        # if new command received, reset event flag to False until gamepad updated.
+        return np.concatenate([delta_pos, rot_vec]), self._close_gripper
+
+    """
+    Internal helpers.
+    """
+
+    def _on_gamepad_event(self, event, *args, **kwargs):
+        """Subscriber callback to when kit is updated.
+
+        Reference:
+            https://docs.omniverse.nvidia.com/kit/docs/carbonite/latest/docs/python/carb.html?highlight=gamepadeventtype#carb.input.Gamepad
+        """
+
+        cur_val = event.value
+        absval = abs(event.value)
+
+        # Ignore 0 since it signifies the movement  of the stick has stopped,
+        # but doesn't mean it's at center...could be being held steady
+
+        if absval < self.deadzone:
+            cur_val = 0
+
+        if event.input == carb.input.GamepadInput.A:
+            if cur_val > 0.5:
+                self._close_gripper = True
+        elif event.input == carb.input.GamepadInput.B:
+            if cur_val > 0.5:
+                self._close_gripper = False
+        elif event.input in self._INPUT_KEY_POS_VALUE_MAPPING:
+            i, j, v = self._INPUT_KEY_POS_VALUE_MAPPING[event.input]
+            self._delta_pos[i, j] = v * cur_val
+        elif event.input in self._INPUT_KEY_ROT_VALUE_MAPPING:
+            i, j, v = self._INPUT_KEY_ROT_VALUE_MAPPING[event.input]
+            self._delta_rot[i, j] = v * cur_val
+        elif event.input in self._INPUT_KEY_ROT_BOOL_MAPPING:
+            i, j, v = self._INPUT_KEY_ROT_BOOL_MAPPING[event.input]
+            if cur_val > 0.5:
+                self._delta_rot[i, j] = v
+            else:
+                self._delta_rot[:, j] = 0
+        elif event.input.name in self._additional_callbacks:
+            self._additional_callbacks[event.input.name]()
+        return True
+
+    def _create_key_bindings(self):
```

Comment:
I propose to rename these variables to make them easier to understand based on the action from the gamepad (stick, dpad, button). What do you think?

* `_INPUT_KEY_POS_VALUE_MAPPING` -> `_INPUT_STICK_VALUE_MAPPING`
* `_INPUT_KEY_ROT_BOOL_MAPPING` -> `_INPUT_DPAD_VALUE_MAPPING`
* `_INPUT_KEY_ROT_VALUE_MAPPING` -> `_INPUT_BUTTON_VALUE_MAPPING`

Reply:
Thanks a lot for this good idea. I changed to dicts to `_INPUT_STICK_VALUE_MAPPING` and `_INPUT_DPAD_VALUE_MAPPING`. I didn't add `_INPUT_BUTTON_VALUE_MAPPING` since now the only button used is `X`.